test/src/middleware/auth.test.js

```diff
@@ -0,0 +1,245 @@
+'use strict';
+
+const expect = require('chai').expect;
+const createError = require('http-errors');
+const sinon = require('sinon');
+const _ = require('lodash');
+
+const auth = require('../../../src/middleware/auth');
+const keysFixtures = require('../../fixtures/auth/keys.fixtures');
+const tokensFixtures = require('../../fixtures/auth/tokens.fixtures');
+
+describe('Authentication middleware - auth.js', function root() {
+  let sandbox;
+
+  before(() => {
+    sandbox = sinon.sandbox.create();
+  });
+
+  beforeEach(() => {
+    sandbox.restore();
+  });
+
+  it('should ignore auth if the config ignoreAuth is true', function* test() {
+    const middleware = auth(true, null);
+    const errorSpy = sandbox.spy(createError);
+    const req = {
+      get: () => {
+      }
+    };
+    const res = {
+      set: () => {
+      }
+    };
+
+    expect(middleware).to.be.instanceof(Function);
+
+    yield middleware(req, res, () => {
+      expect(req.token).to.equal(undefined);
+      expect(req.user).to.equal(undefined);
+      expect(errorSpy.callCount).to.equal(0);
+    });
+  });
+
+  it('should throw error if no key is provided', function* test() {
+    const middleware = auth(false, null);
+    const req = {
+      get: () => {
+      }
+    };
+    const res = {
+      set: () => {
+      }
+    };
+
+    expect(middleware).to.be.instanceof(Function);
+
+    try {
+      yield middleware(req, res, _.noop());
```

Comment:
You want to use the `.noop` reference here not the call of it `_.noop()`. Same below.